vault/cors.go

```diff
@@ -0,0 +1,137 @@
+package vault
+
+import (
+	"errors"
+	"net/http"
+	"regexp"
+	"strings"
+)
+
+var errCORSNotConfigured = errors.New("CORS is not configured")
+
+type CORSConfig struct {
+	enabled          bool
+	allowedOrigins   *[]string
+	allowedHeaders   *map[string]string
```

Comment:
To be clear, I'm not against this if there is a use-case, it just seems outside the scope of the original work which focused on allowed origins and as a result I don't really know what purpose these serve. If there isn't a use-case now, maybe keep the PR focused on one chunk of work and more support could be added later.